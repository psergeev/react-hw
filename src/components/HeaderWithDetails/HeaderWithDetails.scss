header {
    background-color: #222;
    padding: 20px 50px;

    .title {
        color: #b3505d;
        font-weight: bold;
        font-size: 18px;
        margin: 0;
        padding: 0;
    }

    .top-line {
        display: flex;
        justify-content: space-between;

        button {
            cursor: pointer;
            color: #b3505d;
            text-transform: uppercase;
            padding: 10px 60px;
            border: none;
        }
    }

    .details-line {
        margin-top: 30px;
        display: flex;

        img {
            flex: 0 0 250px;
            width: 250px;
            height: 375px;
        }

        .details {
            list-style: none;
            margin: 20px 0 0 50px;
            padding: 0;
            display: inline-block;

            li {
                margin: 10px 0;
            }

            .title {
                color: #E65463;
                font-size: 30px;
            }

            .tagline,
            .overview {
                color: #A9A9A9;
                font-size: 14px;
            }

            .runtime,
            .release-date {
                color: #fff;
                font-weight: bold;
                font-size: 14px;
            }

            .release-date {
                margin-right: 30px;
            }
        }
    }
}
