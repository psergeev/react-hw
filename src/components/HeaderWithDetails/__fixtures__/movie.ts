import { Movie } from '../../../App';

const movie: Movie = {
    poster_path: 'https://image.tmdb.org/t/p/w500/97fNAi62HawGjWru7PvVmF7RAbU.jpg',
    title: 'Kill Bill: Vol. 1',
    tagline: 'Go for the kill.',
    release_date: '2003-10-10',
    vote_average: 7.8,
    runtime: 111,
    overview: 'An assassin is shot at the altar by her ruthless employer, ' +
        'Bill and other members of their assassination circle â€“ but \'The Bride\' ' +
        'lives to plot her vengeance. Setting out for some payback, she makes a death ' +
        'list and hunts down those who wronged her, saving Bill for last.',
    genres: [
        'Action',
        'Crime',
    ],
};

export default movie;
