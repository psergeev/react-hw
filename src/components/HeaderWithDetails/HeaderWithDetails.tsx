import * as React from 'react';
import { Movie } from '../../App';
import VoteMark from '../VoteMark';
import './HeaderWithDetails.scss';

interface Props {
    movie: Movie;
    handleBackToSearchClick: React.MouseEventHandler;
}

export default React.memo((props: Props) => (
    <header>
        <div className="top-line">
            <h1 className="title">netflixroulette</h1>
            <button type="button" onClick={props.handleBackToSearchClick}>BACK TO SEARCH</button>
        </div>
        <div className="details-line">
            <img src={props.movie.poster_path} alt={props.movie.title} />
            <ul className="details">
                <li>
                    <span className="title">{props.movie.title}</span>
                    <VoteMark vote={props.movie.vote_average} />
                </li>
                <li className="tagline">{props.movie.tagline}</li>
                <li>
                    <span className="release-date">{props.movie.release_date.split('-')[0]}</span>
                    <span className="runtime">{`${props.movie.runtime} min`}</span>
                </li>
                <li className="overview">{props.movie.overview}</li>
            </ul>
        </div>
    </header>
));
