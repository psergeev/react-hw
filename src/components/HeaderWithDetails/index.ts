import { connect } from 'react-redux';
import { showSearchAction } from '../../state/actions';
import HeaderWithDetails from './HeaderWithDetails';

export default connect(
    null,
    {
        handleBackToSearchClick: showSearchAction
    }
)(HeaderWithDetails);
