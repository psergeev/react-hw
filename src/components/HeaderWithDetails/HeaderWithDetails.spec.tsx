import * as React from 'react';
import * as renderer from 'react-test-renderer';
import HeaderWithDetails from './HeaderWithDetails';
import movie from './__fixtures__/movie';

jest.mock('../VoteMark', () => 'VoteMark');

test('HeaderWithDetails component works as expected', () => {
    const component = renderer.create(
        <HeaderWithDetails
            onBackToSearchClick={jest.fn()}
            movie={movie}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
