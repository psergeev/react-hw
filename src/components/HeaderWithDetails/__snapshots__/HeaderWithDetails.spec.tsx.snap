// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWithDetails component works as expected 1`] = `
<header>
  <div
    className="top-line"
  >
    <h1
      className="title"
    >
      netflixroulette
    </h1>
    <button
      onClick={[MockFunction]}
      type="button"
    >
      BACK TO SEARCH
    </button>
  </div>
  <div
    className="details-line"
  >
    <img
      alt="Kill Bill: Vol. 1"
      src="https://image.tmdb.org/t/p/w500/97fNAi62HawGjWru7PvVmF7RAbU.jpg"
    />
    <ul
      className="details"
    >
      <li>
        <span
          className="title"
        >
          Kill Bill: Vol. 1
        </span>
        <VoteMark
          vote={7.8}
        />
      </li>
      <li
        className="tagline"
      >
        Go for the kill.
      </li>
      <li>
        <span
          className="release-date"
        >
          2003
        </span>
        <span
          className="runtime"
        >
          111 min
        </span>
      </li>
      <li
        className="overview"
      >
        An assassin is shot at the altar by her ruthless employer, Bill and other members of their assassination circle â€“ but 'The Bride' lives to plot her vengeance. Setting out for some payback, she makes a death list and hunts down those who wronged her, saving Bill for last.
      </li>
    </ul>
  </div>
</header>
`;
